kore/src/Kore/Step/Simplification/Simplify.hs

```diff
@@ -246,11 +247,21 @@ simplifyTerm
         )
     => TermLike variable
     -> simplifier (OrPattern variable)
-simplifyTerm termLike = do
-    TermLikeSimplifier simplify <- askSimplifierTermLike
-    results <- gather $ simplify Predicate.top termLike
-    return (OrPattern.fromPatterns results)
+simplifyTerm = simplifyConditionalTermToOr Predicate.top
 
+{- | Use a 'TermLikeSimplifier' to simplify a pattern subject to conditions.
+ -}
+simplifyConditionalTermToOr
+    :: forall variable simplifier
+    .   ( SimplifierVariable variable
+        , MonadSimplify simplifier
+        )
+    => Predicate variable
+    -> TermLike variable
+    -> simplifier (OrPattern variable)
+simplifyConditionalTermToOr predicate termLike = do
+    results <- gather $ simplifyConditionalTerm predicate termLike
```

Comment:
Does `simplifyConditionalTerm` actually use the predicate now?

Reply:
Yes, it should.